File: Patterns/code/java/DesignPatterns/src/main/java/abstractfactory/AdventureLandFactory.java
package abstractfactory;

class AdventureLandFactory implements AbstractFoodStallFactory {

    @Override
    public IceCream createIceCream() {
        System.out.println("Creating tropical-themed ice cream");
        return new TropicalIceCream();
    }

    @Override
    public Pizza createPizza() {
        System.out.println("Creating exotic-themed pizza");
        return new ExoticPizza();
    }
}

File: code/typescript/src/abstractFactory/AdventureLandFactory.ts
import { AbstractFoodStallFactory } from './AbstractFoodStallFactory';
import { IceCream } from './IceCream';
import { Pizza } from './Pizza';
import { TropicalIceCream } from './TropicalIceCream';
import { ExoticPizza } from './ExoticPizza';

export class AdventureLandFactory implements AbstractFoodStallFactory {
    createIceCream(): IceCream {
        console.log("Creating tropical-themed ice cream");
        return new TropicalIceCream();
    }

    createPizza(): Pizza {
        console.log("Creating exotic-themed pizza");
        return new ExoticPizza();
    }
}
